pipeline/extract.py

```diff
@@ -85,20 +87,72 @@ def get_website_from_url(url: str) -> str:
     return website_url
 
 
-def scrape_from_html(html_content: bytes, url: str, product_id: int) -> dict:
+def scrape_pricing_process(html_content: bytes, url: str, product_id: int) -> dict:
+    """ Chooses which scraper to use based off of the URL """
+
+    website_url = get_website_from_url(url)
+
+    if "https://store.steampowered.com/" in url:
+        return scrape_from_steam_html(html_content, url, product_id)
+
+    if "https://www.amazon.co" in url:
+        return scrape_from_amazon_html(html_content, url, product_id)
+
+    logging.error(
+        "Cannot scrape that URL, since it's not an Amazon/Steam webpage.")
+    return
+
+
+def scrape_from_amazon_html(html_content, url, product_id):
     """ Scrapes from html to get a dictionary with the:
     - Product_ID 
     - product_name
     - original_price
     - discount_price
     - website 
-    """
+    for Amazon Product Pages."""
+    s = BeautifulSoup(html_content, 'html.parser')
+
+    results = s.find("div", id="corePriceDisplay_desktop_feature_div")
+
+    if not results:
+        logging.error("Can't scrape from Amazon URL")
+        return None
+
+    game_title_elem = s.find(id="productTitle")
+
+    if not game_title_elem:
+        logging.error("Cannot find game title on the page for URL: %s", url)
+        return None
+
+    discount_price = results.find(
+        "span", class_="a-price aok-align-center reinventPricePriceToPayMargin priceToPay").text
+    original_price = results.find(
+        "div", class_="a-section a-spacing-small aok-align-center").find("span", class_="a-offscreen").text
+    game_title = game_title_elem.text.strip()
+
+    product_information = {"product_id": product_id,
+                           "original_price": original_price,
+                           "discount_price": discount_price,
+                           "game_title": game_title,
+                           "website": get_website_from_url(url)}
+    return product_information
+
+
+def scrape_from_steam_html(html_content: bytes, url: str, product_id: int) -> dict:
+    """ Scrapes from html to get a dictionary with the:
+    - Product_ID 
+    - product_name
+    - original_price
+    - discount_price
+    - website 
+    for STEAM Games/Products."""
```

Comment:
Can this also be shortened? e.g. "Scrapes product, price and website information from Steam."